### ISSUE TITLE
Crash after Ctrl-S, Ctrl-C

### ISSUE THREAD

**Issue by slingamn** [ergochat/readline#32]
https://github.com/chzyer/readline/issues/163 is still present in our fork

**Comment by slingamn** [comment]
Copying/adapting the repro case:

```go
package main

import (
	"fmt"
	"log"

	"github.com/ergochat/readline"
)

func main() {
	log.SetFlags(log.LstdFlags | log.Lshortfile)

	rl, err := readline.New("> ")
	if err != nil {
		log.Fatal(err)
	}

	line, err := rl.Readline()
	if err != nil {
		log.Fatal(err)
	}
	fmt.Println(line)
}
```

Typing Ctrl-S followed by Ctrl-C will cause (against our v0.0.3):

```
> panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x20 pc=0x4acf55]

goroutine 1 [running]:
github.com/ergochat/readline.(*opSearch).ExitSearchMode(0xc00007e000, 0x18?)
        /data/shivaram/workspace/readline/search.go:129 +0xb5
github.com/ergochat/readline.(*operation).readline(0xc0000640a0, 0x0)
        /data/shivaram/workspace/readline/operation.go:300 +0x585
github.com/ergochat/readline.(*operation).Runes(0xc0000640a0)
        /data/shivaram/workspace/readline/operation.go:430 +0x15a
github.com/ergochat/readline.(*operation).String(...)
        /data/shivaram/workspace/readline/operation.go:397
github.com/ergochat/readline.(*Instance).ReadLine(0x60?)
        /data/shivaram/workspace/readline/readline.go:213 +0x1d
github.com/ergochat/readline.(*Instance).Readline(...)
        /data/shivaram/workspace/readline/readline.go:218
main.main()
        /data/shivaram/workspace/readline/example/crash/crash.go:18 +0x8a
```

**Comment by slingamn** [comment]
Resolved by #33
